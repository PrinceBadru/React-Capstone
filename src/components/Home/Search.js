import React, { useState } from 'react';
import PropTypes from 'prop-types';

const Search = ({ setSearchQuery }) => {
  const [localSearchQuery, setLocalSearchQuery] = useState('');

  return (
    <div className="search-bar">
      <input
        type="text"
        className="search-input"
        placeholder="Search by name"
        value={localSearchQuery}
        onChange={(e) => {
          setLocalSearchQuery(e.target.value);
          setSearchQuery(e.target.value);
        }}
      />
    </div>
  );
};

Search.propTypes = {
  setSearchQuery: PropTypes.func.isRequired,
};

export default Search;
