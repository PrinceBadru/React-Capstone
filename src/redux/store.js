import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import thunk from 'redux-thunk';
import cryptoReducer from './Crypto/CryptoSlice';

const middleware = [...getDefaultMiddleware(), thunk];

const store = configureStore({
  reducer: {
    crypto: cryptoReducer,
  },
  middleware,
});

export default store;
