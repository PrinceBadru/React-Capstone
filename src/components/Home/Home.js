import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { Card } from 'react-bootstrap';
import { FaArrowCircleRight } from 'react-icons/fa';
import { fetchCrypto } from '../../redux/Crypto/CryptoSlice';
import '../../App.css';
import Search from './Search';

const Home = () => {
  const dispatch = useDispatch();
  const cryptos = useSelector((state) => state.crypto.data.coins);
  const [searchQuery, setSearchQuery] = useState('');

  useEffect(() => {
    dispatch(fetchCrypto(searchQuery));
  }, [dispatch, searchQuery]);

  const filteredCryptos = cryptos && cryptos.filter((crypto) => {
    const lowerCaseName = crypto.name.toLowerCase();
    const lowerCaseSearchQuery = searchQuery.toLowerCase();
    return lowerCaseName.includes(lowerCaseSearchQuery);
  });

  return (
    <div className="home-div">
      <Search setSearchQuery={setSearchQuery} />
      <div className="status-nav">
        <div className="nav-items">
          <span className="Page-title">MarketCap Rank</span>
        </div>
      </div>
      <div className="card-section">
        {filteredCryptos && filteredCryptos.length > 0 ? (
          filteredCryptos.map((crypto, index) => (
            <div style={{ width: '50%' }} key={crypto.id}>
              <Card
                className="p-2"
                style={{
                  backgroundColor: index % 2 === 0 ? 'red' : 'lightred',
                  color: 'white',
                  padding: '7px',
                  height: '96%',
                  border: '1px solid black',
                }}
              >
                <Card.Header className="header-card">
                  <Card.Title className="card-title">{crypto.name}</Card.Title>
                  <div className="div-cardImg">
                    <Card.Img
                      variant="top"
                      src={crypto.iconUrl}
                      alt="Coin Icon"
                      style={{ width: '70%', height: '70%', objectFit: 'contain' }}
                    />
                  </div>
                  <Link to={`/details/${crypto.name}`}>
                    <FaArrowCircleRight className="details-nav" />
                  </Link>
                </Card.Header>
                <Card.Body className="cardBody">
                  <Card.Subtitle className=" text-muted">
                    Market Cap:
                    {' '}
                    {crypto.marketCap}
                  </Card.Subtitle>
                  <Card.Subtitle className=" text-muted">
                    Price:
                    {' '}
                    {crypto.price}
                  </Card.Subtitle>
                  <Card.Subtitle className=" text-muted">
                    Change:
                    {' '}
                    {crypto.listedAt}
                  </Card.Subtitle>
                </Card.Body>
              </Card>
            </div>
          ))
        ) : (
          <div className="loading-text">Loading Data Please Wait...</div>
        )}
      </div>
    </div>
  );
};

export default Home;
