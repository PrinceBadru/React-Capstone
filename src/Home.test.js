import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import Home from './components/Home/Home';
import store from './redux/store';

test('renders Home component correctly', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <Router>
        <Home />
      </Router>
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
