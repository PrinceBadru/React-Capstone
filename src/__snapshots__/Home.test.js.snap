// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home component correctly 1`] = `
<DocumentFragment>
  <div
    class="home-div"
  >
    <div
      class="search-bar"
    >
      <input
        class="search-input"
        placeholder="Search by name"
        type="text"
        value=""
      />
    </div>
    <div
      class="status-nav"
    >
      <div
        class="nav-items"
      >
        <span
          class="Page-title"
        >
          Status by Crypto
        </span>
      </div>
    </div>
    <div
      class="card-section"
    >
      <div
        class="loading-text"
      >
        Loading Data Please Wait...
      </div>
    </div>
  </div>
</DocumentFragment>
`;
